Usecases for publishing data (#53)
ElektraInitiative/opensesame: doc/usecases/assigned/public_publishing_of_data_for_anyone_interested.md

# Use Case: public publishing of data for anyone interested

## Summary

> - **Scope:** Clima-Sensor-US - Module

Reviewer: ```suggestion
- **Scope:** Clima-Sensor-US
```

> - **Level:** Backend

Reviewer: ```suggestion
- **Level:** User Goal
```

> - **Actors** Thies-Clima-Sensor, Opensensemap

Reviewer: Opensesame is probably also involved.

```suggestion
- **Actors** Thies-Clima-Sensor, Opensesame, Opensensemap
```

- **Brief:** The sensor informations are collected periodically and are published to opensensemap
- **Assignee:** Felix
- **Status:** Assigned

## Scenarios

- **Precondition:** 

> 	- The user has configured opensesame to use sensor mode.

Reviewer: This is only temporary workaround and hopefully is not needed long, so instead it should be:

```suggestion
	- The user has configured opensesame to use Clima-Sensor-US  and to publish to opensensemap.
```

> 	- Sensor IDs are set in constants.

Author: We used constants to store the IDs of the sensors, which are directly in the source code.

> 	- Sensor IDs are set in constants.

Reviewer: Then this is not a precondition for a use case.

```suggestion
```

Please undraft, so that we can get this merged.

> 	- Update frequency is set.

Reviewer: Not relevant for user, it is fixed with a minute.

Author: Ok, so the update frequency should not be set by the user?

Reviewer: I don't see why it is needed. The web interface seems to use hard-coded 1 minute, or can this be changed? But even if, I don't really see why I would want to change it.
In general: if there is no need, we always go with the minimal solution and keep things hard-coded.

- **Main Success Scenario:** 
	- Retrieve the register values of the clima sensor.
	- Generate a JSON format using sensor IDs and register values.

> 	- Transmit the JSON string to the Opensensemap API.

Reviewer: ```suggestion
	- Opensesame transmits the JSON string to the Opensensemap API every minute.
```
